src/middleware/log_request.rs

```diff
@@ -31,10 +31,16 @@ impl Handler for LogRequests {
         let response_time =
             response_time.as_secs() * 1000 + u64::from(response_time.subsec_nanos()) / 1_000_000;
 
+        let metadata_length = req
+            .extensions()
+            .find::<u64>()
+            .map_or(String::new(), |l| format!(" metadata_length={}", l));
+
         print!(
             "at={level} method={method} path=\"{path}\" \
              request_id={request_id} fwd=\"{ip}\" service={time_ms}ms \
-             status={status} user_agent=\"{user_agent}\"",
+             status={status} user_agent=\"{user_agent}\"\
+             metadata_length={metadata_length}",
```

Comment:
When I test this out locally, I'm seeing this on requests that don't have a `metadata_length`:

```
at=info method=GET path="/api/v1/summary" request_id= fwd="" service=44ms status=200 user_agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Gecko/20100101 Firefox/72.0"metadata_length=
```

And this on requests that do:

```
at=info method=PUT path="/api/v1/crates/new" request_id= fwd="" service=1060ms status=200 user_agent="cargo 1.39.0 (1c6ec66d5 2019-09-30)"metadata_length= metadata_length=734
```

I think we can remove the `metadata_length=` here now because it's duplicated with [this](https://github.com/rust-lang/crates.io/pull/1905/files#diff-3c16efcf7a504a2aae66b1d607874710R37) ` metadata_length=`.

Could you make that change please? Once that's done, I think this is good to go! And don't worry about the commits, I can fix them up 👍

Reply:
I definitely should have noticed that, thanks.